### PR TITLE
Add a stat for security guards: Vandals stopped

### PR DESCRIPTION
**Pull request by manatails**
I propose adding a stat for security guards, currently the info panel gives zero information about what they do. Why don't we add a little stat for security guards so we know how many vandals they actually stopped?

### CODE REVIEW

**Comment by IntelOrca** [comment]
This requires a new save format, otherwise this stat will reset every time you load a saved game.

**Comment by manatails** [comment]
> 
> 
> This requires a new save format, otherwise this stat will reset every time you load a saved game.

I made the variable a union with StaffLitterSwept, it will be compatible with older saves and will be persistent across different saves

**Review by ZehMatt — APPROVED** [review]
Code looks alright, should be also ok with with SV6 compatibility. When new sprites are created everything is zeroed so this field for security peeps should be usually zero in older saves.

File: src/openrct2/peep/Guest.cpp [hunk 1]
```diff
@@ -5945,7 +5945,11 @@ static void peep_update_walking_break_scenery(Peep* peep)
         int32_t y_diff = abs(inner_peep->y - peep->y);
 
         if (std::max(x_diff, y_diff) < 224)
+        {
+            inner_peep->StaffVandalsStopped++;
+
```
Comment (tupaschoal) [inline src/openrct2/peep/Guest.cpp:5950]:
```suggestion
```

Seems like a weird line break

File: src/openrct2-ui/windows/Staff.cpp [hunk 2]
```diff
@@ -1139,6 +1139,12 @@ void window_staff_stats_paint(rct_window* w, rct_drawpixelinfo* dpi)
             gfx_draw_string_left(dpi, STR_STAFF_STAT_RIDES_FIXED, ft.Data(), COLOUR_BLACK, screenCoords);
             break;
         case StaffType::Security:
+            ft = Formatter();
+            ft.Add<uint16_t>(peep->StaffVandalsStopped);
+            gfx_draw_string_left(dpi, STR_STAFF_STAT_VANDALS_STOPPED, ft.Data(), COLOUR_BLACK, screenCoords);
+            screenCoords.y += LIST_ROW_HEIGHT;
+
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/Staff.cpp:1146]:
```suggestion
```

Seems like a weird line break

**Review by tupaschoal — COMMENTED** [review]
Can you please add a screenshot of how it looks before and after?

Thanks!

**Comment by manatails** [comment]
> 
> 
> Can you please add a screenshot of how it looks before and after?
> 
> Thanks!

2021-02-14 10_14_32-OpenRCT2
Before

2021-02-14 10_20_36-OpenRCT2
After

File: src/openrct2-ui/windows/Staff.cpp [hunk 3]
```diff
@@ -1139,6 +1139,11 @@ void window_staff_stats_paint(rct_window* w, rct_drawpixelinfo* dpi)
             gfx_draw_string_left(dpi, STR_STAFF_STAT_RIDES_FIXED, ft.Data(), COLOUR_BLACK, screenCoords);
             break;
         case StaffType::Security:
+            ft = Formatter();
+            ft.Add<uint16_t>(peep->StaffVandalsStopped);
+            gfx_draw_string_left(dpi, STR_STAFF_STAT_VANDALS_STOPPED, ft.Data(), COLOUR_BLACK, screenCoords);
+            screenCoords.y += LIST_ROW_HEIGHT;
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/Staff.cpp:1145]:
```suggestion
```
No need for this line, as it is the last line in the tab.

**Review by tupaschoal — APPROVED** [review]

**Review by Gymnasiast — COMMENTED** [review]
Code looks good. Please add a changelog entry and bump the network version, and we'll merge it.

`- Feature: [#14071] “Vandals stopped” statistic for security guards.`

**Comment by manatails** [comment]
Prepared it for merge

**Review by tupaschoal — APPROVED** [review]

**Pull request merged**